### PR TITLE
Merge modernized code from trunk

### PR DESCRIPTION
**Pull request by barijaona**
Code modernization and warnings fixes,
mostly made by @nivekkagicom (B. Kevin Hardman)

Conflicts solved : MMTabBarButton.m, MMTabBarController.m,
MMTabBarView.m, MMMojaveTabStyle+Assets.m

### CODE REVIEW

File: Configurations/Base.xcconfig [hunk 1]
```diff
@@ -11,3 +11,4 @@ GCC_PREFIX_HEADER = $(SYSTEM_LIBRARY_DIR)/Frameworks/AppKit.framework/Headers/Ap
 GCC_PRECOMPILE_PREFIX_HEADER = YES
 CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES
 CLANG_ENABLE_MODULES = YES
+WARNING_CFLAGS = -Weverything -Wno-custom-atomic-properties -Wno-direct-ivar-access -Wno-float-equal -Wno-gnu  -Wno-objc-missing-property-synthesis
```
Comment (barijaona) [inline Configurations/Base.xcconfig]: Subtle change

File: MMTabBarView/MMTabBarView/MMOverflowPopUpButton.m [hunk 2]
```diff
@@ -223,12 +220,16 @@ - (void)_stopCellAnimation {
 }
 
 - (CGFloat)secondImageAlpha {
-    return [[self cell] secondImageAlpha];
+    return [self.cell secondImageAlpha];
 }
 
 - (void)setSecondImageAlpha:(CGFloat)value {
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMOverflowPopUpButton.m]: Subtle change

File: MMTabBarView/MMTabBarView/MMTabBarView.m [hunk 3]
```diff
@@ -1724,81 +1728,82 @@ - (NSDragOperation)draggingEntered:(id <NSDraggingInfo>)sender {
...
-    NSPasteboard *pb = [sender draggingPasteboard];
+    NSPasteboard *pb = sender.draggingPasteboard;
     
     if ([pb canReadItemWithDataConformingToTypes:[NSArray arrayWithObject:AttachedTabBarButtonUTI]]) {
         
-        MMTabDragAssistant *dragAssistant = [MMTabDragAssistant sharedDragAssistant];
+        MMTabDragAssistant *dragAssistant = MMTabDragAssistant.sharedDragAssistant;
         return [dragAssistant draggingUpdated:sender inTabBarView:self];
          
-    } else {
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMTabBarView.m]: Need to check if the modified logic leads to same behavior

File: MMTabBarView/MMTabBarView/MMTabBarView.m [hunk 4]
```diff
@@ -1993,10 +1998,18 @@ - (void)fireSpring:(NSTimer *)timer {
 #pragma mark Menu Validation
 
 - (BOOL)validateMenuItem:(NSMenuItem *)sender {
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMTabBarView.m]: Check if the result is same

File: MMTabBarView/MMTabBarView/MMTabDragAssistant.m [hunk 5]
```diff
@@ -85,110 +85,121 @@ - (NSDragOperation)draggingSession:(NSDraggingSession *)session sourceOperationM
...
+	NSTabView *sourceTabView = _sourceTabBar.tabView;
+	NSUInteger sourceIndex = pasteboardItem.sourceIndex;
 
-    if ([self isDragging]) {   // means there was not a successful drop (performDragOperation)
+    if (self.isDragging) {   // means there was not a successful drop (performDragOperation)
     
-        id <MMTabBarViewDelegate> sourceDelegate = [_sourceTabBar delegate];
+        id <MMTabBarViewDelegate> sourceDelegate = _sourceTabBar.delegate;
 
         //split off the dragged tab into a new window
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMTabDragAssistant.m]: Modified logic here

File: MMTabBarView/MMTabBarView/MMTabDragAssistant.m [hunk 6]
```diff
@@ -369,20 +383,24 @@ - (BOOL)performDragOperation:(id <NSDraggingInfo>)sender forTabBarView:(MMTabBar
 
 - (void)finishDragOfPasteboardItem:(MMTabPasteboardItem *)pasteboardItem {
 
-    NSTabView *sourceTabView = [_sourceTabBar tabView];
-	MMTabBarView *destinationTabBarView = [self destinationTabBar];
-    
-	if ([sourceTabView numberOfTabViewItems] == 0 && [[_sourceTabBar delegate] respondsToSelector:@selector(tabView:closeWindowForLastTabViewItem:)]) {
-		[[_sourceTabBar delegate] tabView:sourceTabView closeWindowForLastTabViewItem:[_attachedTabBarButton tabViewItem]];
+	MMAttachedTabBarButton* const tabBarButton = _attachedTabBarButton;
+	if (tabBarButton == nil) {
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMTabDragAssistant.m]: Subtle change

File: MMTabBarView/MMTabBarView/Styles/Mojave Tab Style/MMMojaveTabStyle+Assets.m [hunk 7]
```diff
@@ -53,35 +57,19 @@ - (id)assetForPart:(MMMojaveTabStyleAsset)part ofTabBarView:(MMTabBarView *)tabB
     else
     {
 #endif
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/Styles/Mojave Tab Style/MMMojaveTabStyle+Assets.m]: Modified logic

File: MMTabBarView/MMTabBarView/MMOverflowPopUpButton.m [hunk 8]
```diff
@@ -223,12 +220,16 @@ - (void)_stopCellAnimation {
...
-    [[self cell] setSecondImageAlpha:value];
-    [self updateCell:[self cell]];
+	MMOverflowPopUpButtonCell* const cell = self.cell;
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMOverflowPopUpButton.m]: Subtle change

File: MMTabBarView/MMTabBarView/MMTabBarView.m [hunk 9]
```diff
@@ -1993,10 +1998,18 @@ - (void)fireSpring:(NSTimer *)timer {
...
-	[sender setState:([[sender representedObject] isEqualTo:[_tabView selectedTabViewItem]]) ? NSOnState : NSOffState];
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/MMTabBarView.m]: need to check if final result is the same

File: MMTabBarView/MMTabBarView/Styles/Mojave Tab Style/MMMojaveTabStyle+Assets.m [hunk 10]
```diff
@@ -53,35 +57,19 @@ - (id)assetForPart:(MMMojaveTabStyleAsset)part ofTabBarView:(MMTabBarView *)tabB
...
-        }
-        else
-        {
-            if ( [tabBarView isWindowActive ] )
-                mode = MMMappearanceAquaLight;
-            else
-                mode = MMMappearanceAquaLightInactive;
-        }
+		if ( @available(macos 10.10, *) ) {
+			if ( NSWorkspace.sharedWorkspace.accessibilityDisplayShouldIncreaseContrast )
```
Comment (barijaona) [inline MMTabBarView/MMTabBarView/Styles/Mojave Tab Style/MMMojaveTabStyle+Assets.m]: modified logic

File: MMTabBarView/MMTabBarView/MMTabBarView.m (hunk 3 above)
Comment (barijaona) [reply MMTabBarView/MMTabBarView/MMTabBarView.m]: OK for me

File: MMTabBarView/MMTabBarView/MMTabBarView.m (hunk 9 above)
Comment (barijaona) [reply MMTabBarView/MMTabBarView/MMTabBarView.m]: Safety measures seem OK to me

File: MMTabBarView/MMTabBarView/MMTabDragAssistant.m (hunk 5 above)
Comment (barijaona) [reply MMTabBarView/MMTabBarView/MMTabDragAssistant.m]: OK for me

File: MMTabBarView/MMTabBarView/MMTabDragAssistant.m (hunk 6 above)
Comment (barijaona) [reply MMTabBarView/MMTabBarView/MMTabDragAssistant.m]: OK for me

File: MMTabBarView/MMTabBarView/MMTabDragAssistant.m (hunk 5 above)
Comment (TAKeanice) [reply MMTabBarView/MMTabBarView/MMTabDragAssistant.m]: This one is identical, just using dot syntax, isn´t it?

Comment (barijaona) [reply MMTabBarView/MMTabBarView/MMTabDragAssistant.m]: Yes the above is identical. But my little concern was with the following lines. OK for me now.

**Pull request merged**